docs/semgrep-secrets/validators.md

```diff
@@ -0,0 +1,327 @@
+  - regex
+  validators:
+  - http:
+      request:
+        headers:
+          Authorization: Bearer $REGEX
+          Host: api.exampleCo.com
+          User-Agent: Semgrep
+        method: GET
+        url: https://api.exampleCo.com/user
+      response:
+      - match:
+        - status-code: 200
+        result:
+          validity: valid
+      - match:
+        - status-code: 401
+        result:
+          validity: invalid 
+  patterns:
+  - patterns:
+    - pattern-regex: (?<REGEX>\b(someprefix_someRegex[0-9A-Z]{32})\b)
+    - focus-metavariable: $REGEX
+    - metavariable-analysis:
+        analyzer: entropy
+        metavariable: $REGEX
+```
+
+</details>
+
+## Syntax
+
+### validator
+
+| Key | Required | Description |
+| - | - | - |
+| validator | Yes | Used to define a validator within a Semgrep rule. |
+
+### http
+
+| Key | Required | Description |
+| - | - | - |
+| http | Yes | Indicates that the request type is `http`. |
+
+### request
+
+| Key | Required | Description |
+| - | - | - |
+| request | Yes | Describes the request object and the URL to which the request object should be sent |
+| method | Yes | The HTTP method Semgrep uses to make the call. Accepted values: `GET`, `POST`, `PUT`, `DELETE`, `OPTIONS`, `PATCH` |
+| url | Yes | The URL to which the call is made |
+| headers | Yes | The headers to include with the call |
+| body | No | The body used with `POST` requests |
+
+#### Sub-keys for `headers`
+
+The following keys are for use with `headers`:
+
+| Key | Required | Description |
+| - | - | - |
+| Host | Yes | The host to which the call is made |
+| Other-values | No | The request header. Accepts all values, including `Authorization`, `Content-Type`, `User-Agent`, and so on  |
+
+#### Example
+
+```yaml
+request:
+  headers:
+    Authorization: Bearer $REGEX
+    Host: api.exampleCo.com
+    User-Agent: Semgrep
+  method: GET
+  url: https://api.exampleCo.com/user
+```
+
+### response
+
+| Key | Required | Description |
+| - | - | - |
+| response | Yes | Describes the response object expected and how to handle the results returned |
+| match | Yes | Defines the list of match conditions. |
+
+#### Sub-keys for `match`
+
+| Key | Required | Description |
+| - | - | - |
+| status-code | Yes | The HTTP status code expected by Semgrep Secrets for it to consider the secret a match |
+| result | Yes | Defines the result of the call based on the HTTP status code received |
+| content | No | The response body; you can inspect it for a specific value or JSON response to determine if the request is valid. An example of where this is useful is when both invalid and valid responses return the same status code |
+| message | No | Used to override the rule message based on the secret's validity state |
+| metadata | No | Used to override existing metadata fields or add new metadata fields based on the secret's validity state |
+| severity |  No | Used to override the existing rule severity based on the validity state |
```

Comment:
These are the fields of the object which is the value corresponding to the `result` key.